artifacts.go

```diff
@@ -58,7 +60,13 @@ func artifactsHandler(indexer Indexer, cacheTime time.Duration) func(w http.Resp
 			return
 		}
 
+		aPackage := packageList[0]
+		if storageIndexer, assert := aPackage.Indexer().(*storage.Indexer); assert {
+			storageIndexer.HijackArtifactsHandler(w, r, aPackage)
```

Comment:
I wouldn't call it "Hijack", here we are doing a legit redirect :slightly_smiling_face:

Some suggestions:
```suggestion
			storageIndexer.RedirectArtifactsHandler(w, r, aPackage)
```
```suggestion
			storageIndexer.RemoteArtifactsHandler(w, r, aPackage)
```

Reply:
I can apply those renamings, not a problem :)

I found an analogy with [http.Hijacker](https://pkg.go.dev/net/http#Hijacker), which lets a handler take over the TCP connection. In this case, we let Storage Indexer handlers process requests, which include a redirection.